src/persist/src/operators/stream.rs

```diff
@@ -893,31 +892,33 @@ mod tests {
             Condition(u64),
         }
 
-        let (listen_tx, listen_rx) = mpsc::channel();
+        let (listen_primary_tx, listen_primary_rx) = crossbeam_channel::unbounded();
+        primary_read.listen(listen_primary_tx)?;
+
+        let (listen_condition_tx, listen_condition_rx) = crossbeam_channel::unbounded();
+        condition_read.listen(listen_condition_tx)?;
 
-        {
-            let listen_tx = listen_tx.clone();
-            let listen_fn = ListenFn(Box::new(move |e| {
-                match e {
-                    ListenEvent::Sealed(t) => listen_tx.send(Sealed::Primary(t)).unwrap(),
+        // best effort loop to categorize and forward incoming messages to an aggregate channel
+        // for later test validation logic
+        let (listen_aggregate_tx, listen_aggregate_rx) = crossbeam_channel::unbounded();
+        let _ = thread::spawn(move || loop {
```

Comment:
let's use an async runtime for this, so we can use something like https://docs.rs/futures/0.3.17/futures/macro.select.html to repeatedly grab from whichever is available first without the hot loop. we should also have that bit wait until both channels are closed before returning and block the test on this async task finished to ensure everything is cleaned up at the end